From ed427251b703dc832a2051b5e4bdd1b51b99b115 Mon Sep 17 00:00:00 2001
From: Guido Günther <agx@sigxcpu.org>
Date: Wed, 25 Mar 2020 11:59:03 +0100
Subject: [PATCH] proximity: Get near-level from sysfs if available
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

DT based devices can set the near-level via sysfs. Parse it from
there too.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Closes: #297
---
 src/drv-iio-poll-proximity.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/drv-iio-poll-proximity.c b/src/drv-iio-poll-proximity.c
index 98bacb1..432cda5 100644
--- a/src/drv-iio-poll-proximity.c
+++ b/src/drv-iio-poll-proximity.c
@@ -100,6 +100,9 @@ get_near_level (GUdevDevice *device)
 	gint near_level;
 
 	near_level = g_udev_device_get_property_as_int (device, PROXIMITY_NEAR_LEVEL);
+	if (!near_level)
+		near_level = g_udev_device_get_sysfs_attr_as_int (device, "in_proximity_nearlevel");
+
 	if (!near_level) {
 		g_warning ("Found proximity sensor but no " PROXIMITY_NEAR_LEVEL " udev property");
 		g_warning ("See https://gitlab.freedesktop.org/hadess/iio-sensor-proxy/blob/master/README.md");
-- 
2.30.0

